version: '3.7'

services:
  proxyer:
    image: dockerhub.datagrand.com/yskg/nginx:1.18.0
    hostname: proxyer
    ports:
      - 50070:50070
      - 8088:8088
      - 8080:8080
      - 8188:8188
      - 16010:16010
      - 16031:16031
      - 16032:16032
      - 33066:3306
      - 8183:8182     # janus
    deploy:
      endpoint_mode: vip
    volumes:
      - ./data/hadoop/conf/hadoop.conf:/etc/nginx/conf.d/hadoop.conf
      - ./data/spark/conf/spark.conf:/etc/nginx/conf.d/spark.conf
      - ./data/hbase/conf/hbase.conf:/etc/nginx/conf.d/hbase.conf
      - ./data/proxyer/yskg.conf:/etc/nginx/conf.d/yskg.conf
      - ./data/proxyer/nginx.conf:/etc/nginx/nginx.conf
    networks:
      - bigdata
    healthcheck:
      test: ["CMD-SHELL", "curl --fail localhost || exit 1"]
      interval: 30s
      timeout: 10s
      retries: 3

networks:
  bigdata:
    external: true
    name: xnet
